### PR TITLE
Added callback feature

### PR DESCRIPTION
**Pull request by alexschimpf**
## Pull Request Checklist

- [X] Make sure to read the contributor guide [here](https://github.com/alexschimpf/fastapi-versionizer/tree/main/CONTRIBUTE.md).
- [X] Make sure you are making a pull request against the main branch.
- [X] Make sure your pull request title matches the requested format.
- [X] Make sure to lint, type-check, and run unit and API tests.

## Description
Added a new `callback` param which fires when a new versioned app is created, and before it is mounted.

### CODE REVIEW

File: examples/http_basic_auth.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            openapi_url=openapi_url,
+            title=title
+        )
+
+
+'''
+We call `versionize` but do not tell it to generate any docs/redoc routes.
+Using the `callback`, we instead generate our own docs/redoc routes with HTTP basic auth.
+'''
+versions = versionize(
```
Comment (ollz272) [inline examples/http_basic_auth.py:85]: I suppose it would be good to have some tests for this example

**Review by ollz272 — APPROVED** [review]
Looks great :-) Just maybe some tests needed?

File: examples/http_basic_auth.py (hunk 1 above)
Comment (alexschimpf) [reply examples/http_basic_auth.py:71]: Added!

**Pull request merged**